platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java

```diff
@@ -342,28 +335,125 @@ public NodeId getCreatorId() {
         return creatorId;
     }
 
+    /**
+     * Get the address book round of the event.
+     *
+     * @return the address book round of the event
+     */
+    public long getAddressBookRound() {
+        return addressBookRound;
+    }
+
+    /**
+     * Get the event descriptor for the self parent.
+     *
+     * @return the event descriptor for the self parent
+     */
+    public EventDescriptor getSelfParent() {
+        return selfParent;
+    }
+
+    /**
+     * Get the event descriptors for the other parents.
+     *
+     * @return the event descriptors for the other parents
+     */
+    public List<EventDescriptor> getOtherParents() {
+        return otherParents;
+    }
+
+    /**
+     * Get the self parent generation
+     *
+     * @return the self parent generation
+     */
     public long getSelfParentGen() {
-        return selfParentGen;
+        if (selfParent == null) {
+            return -1;
```

Comment:
use `EventConstants.GENERATION_UNDEFINED`

Reply:
Or we just use common and move the event related implementations and constants to common.

Reply:
I would say move the constants to common

Reply:
will do